src/south/SQLDbToFile/SQLDbToFile.class.js

```diff
@@ -228,97 +233,125 @@ class SQLDbToFile extends ProtocolHandler {
+      connexion = new Client(config)
+      await connexion.connect()
       const params = this.containsLastCompletedDate ? [new Date(this.lastCompletedAt)] : []
-      const [rows] = await connection.execute(
-        { sql: adaptedQuery, timeout: this.requestTimeout },
-        params,
-      )
+      const { rows } = await connexion.query(adaptedQuery, params)
       data = rows
     } catch (error) {
       this.logger.error(error)
     } finally {
-      if (connection) {
-        await connection.end()
-      }
+      await connexion.end()
     }
 
     return data
   }
 
   /**
-   * Get new entries from PostgreSQL database.
+   * Get new entries from Oracle database.
    * @returns {void}
    */
-  async getDataFromPostgreSQL() {
-    const adaptedQuery = this.query.replace('@LastCompletedDate', '$1')
+  async getDataFromOracle() {
+    const adaptedQuery = this.query.replace('@LastCompletedDate', ':date1')
     this.logger.debug(`Executing "${adaptedQuery}" ${this.containsLastCompletedDate ? 'with' : 'without'} LastCompletedDate`)
 
     const config = {
-      host: this.host,
-      port: this.port,
       user: this.username,
       password: this.encryptionService.decryptText(this.password),
-      database: this.database,
-      query_timeout: this.requestTimeout,
+      connectString: `${this.host}:${this.port}/${this.database}`,
     }
 
-    types.setTypeParser(1114, (str) => new Date(`${str}Z`))
-    let connexion = null
+    let connection = null
     let data = []
     try {
-      connexion = new Client(config)
-      await connexion.connect()
+      process.env.ORA_SDTZ = 'UTC'
+      oracledb.outFormat = oracledb.OUT_FORMAT_OBJECT
+      connection = await oracledb.getConnection(config)
+      connection.callTimeout = this.requestTimeout
       const params = this.containsLastCompletedDate ? [new Date(this.lastCompletedAt)] : []
-      const { rows } = await connexion.query(adaptedQuery, params)
+      const { rows } = await connection.execute(adaptedQuery, params)
       data = rows
     } catch (error) {
       this.logger.error(error)
     } finally {
-      if (connexion) {
-        await connexion.end()
+      if (connection) {
+        await connection.close()
       }
     }
 
     return data
   }
 
   /**
-   * Get new entries from Oracle database.
+   * Get new entries from local SQLite db file
    * @returns {void}
    */
-  async getDataFromOracle() {
-    const adaptedQuery = this.query.replace('@LastCompletedDate', ':date1')
+  async getDataFromSqlite() {
+    const adaptedQuery = this.query.replace('@LastCompletedDate', '?')
+    this.logger.debug(`Executing "${adaptedQuery}" ${this.containsLastCompletedDate ? 'with' : 'without'} LastCompletedDate`)
+
+    let database = null
+    let data = []
+    try {
+      database = await sqlite.open({ filename: this.databasePath, driver: sqlite3.Database })
+      const stmt = await database.prepare(adaptedQuery)
+      if (this.containsLastCompletedDate) {
```

Comment:
I like one-liners :}
`const preparedParameters = this.containsLastCompletedDate ? { '@LastCompletedDate': new Date(this.lastCompletedAt) } : {}`
`data = await stmt.all(preparedParameters)`